omod/src/test/java/org/openmrs/module/idgen/rest/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -0,0 +1,145 @@
+package org.openmrs.module.idgen.rest.controller;
+
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.PatientIdentifierType;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.idgen.AutoGenerationOption;
+import org.openmrs.module.idgen.service.IdentifierSourceService;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+/**
+ * Tests CRUD operations for {@link AutoGenerationOption}s via web service calls
+ */
+public class AutoGenerationOptionControllerTest extends MainResourceControllerTest {
+	
+	@Autowired
+	private IdentifierSourceService identifierSourceService;
+	
+	public PatientIdentifierType patientIdentifierType;
+	
+	@Before
+	public void before() throws Exception {
+		executeDataSet("org/openmrs/module/idgen/include/TestData.xml");
+	}
+	
+	@Override
+	public String getURI() {
+		return "idgen/autogenerationoption";
+	}
+	
+	@Override
+	public String getUuid() {
+		return "2f470aa8-1d73-43b7-81b5-01f0c0dfa53c";
```

Comment:
Declare constants at the top of the class or in one file. final string IDENTIFIER_TYPE_UUID = "2f470aa8-1d73-43b7-81b5-01f0c0dfa53c"

Reply:
okay will do that